driver/src/contracts/stablex_contract_auction_element.rs

```diff
@@ -0,0 +1,224 @@
+use byteorder::{BigEndian, ByteOrder};
+
+use std::collections::HashMap;
+
+use web3::types::{H160, U256};
+
+use dfusion_core::models::{BatchInformation, Order};
+
+#[derive(Debug, Default, PartialEq)]
+pub struct AuctionElement {
+    valid_from: U256,
+    valid_until: U256,
+    pub sell_token_balance: u128,
+    pub order: Order,
+}
+
+impl AuctionElement {
+    pub fn in_auction(&self, index: U256) -> bool {
+        self.valid_from < index && index <= self.valid_until
```

Comment:
Just saw in the other PR, this should be <= on both sides.

Reply:
If so, then one of the unit tests will have to be changed. `in_auction_left` should have the left value shifted by one.